h2o-core/src/main/java/water/api/FlowLoadServlet.java

```diff
@@ -0,0 +1,79 @@
+package water.api;
+
+import water.H2O;
+import water.JettyHTTPD;
+import water.persist.PersistManager;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.InputStream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FlowLoadServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+    String uri = JettyHTTPD.getDecodedUri(request);
+    try {
+      Pattern p = Pattern.compile(".*/FlowLoad.bin/flow_dir");
+      Matcher m = p.matcher(uri);
+      if (!m.matches()) {
+        JettyHTTPD.setResponseStatus(response, HttpServletResponse.SC_BAD_REQUEST);
+        response.getWriter().write("Improperly formatted URI");
+        return;
+      }
+      JettyHTTPD.setResponseStatus(response, HttpServletResponse.SC_OK);
+
+      String responsePayload = "{ \"flow_dir\" : \"" + H2O.ARGS.flow_dir + "\" }\n";
+      response.setContentType("application/json");
+      response.getWriter().write(responsePayload);
+    } catch (Exception e) {
+      JettyHTTPD.sendErrorResponse(response, e, uri);
+    } finally {
+      JettyHTTPD.logRequest("GET", request, response);
+    }
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+    String uri = JettyHTTPD.getDecodedUri(request);
+    try {
+      Pattern p = Pattern.compile(".*/FlowLoad.bin/([^/]+)/([^/]+)");
+      Matcher m = p.matcher(uri);
+      boolean b = m.matches();
+      if (!b) {
+        JettyHTTPD.setResponseStatus(response, HttpServletResponse.SC_BAD_REQUEST);
+        response.getWriter().write("Improperly formatted URI");
+        return;
+      }
+
+      String categoryName = m.group(1);
+      String keyName = m.group(2);
+
+      String source = request.getParameter("path");
+      PersistManager persistManager = H2O.getPM();
+
+      InputStream is = persistManager.open(source);
+      if (is == null) {
+        return;
+      }
+      H2O.getNPS().put(categoryName, keyName, is);
```

Comment:
IMHO belongs somewhere else. 

This just copies data from a storage to the NodePersistantStorage. I don't think it warrants a servlet. Servlets are more for binary data that cannot be handled elsewhere.